sdks/java/testing/test-utils/build.gradle

```diff
@@ -39,3 +39,14 @@ dependencies {
   testCompile library.java.hamcrest_library
   testRuntimeOnly project(path: ":runners:direct-java", configuration: "shadowTest")
 }
+
+task verifyJavaVersion(type: Test) {
+  filter {
+    includeTestsMatching 'org.apache.beam.sdk.testutils.jvmverification.JvmVerification.verifyCodeIsCompiledWithJava8'
+    includeTestsMatching 'org.apache.beam.sdk.testutils.jvmverification.JvmVerification.verifyTestCodeIsCompiledWithJava8'
```

Comment:
We should add test that compiles beam with java8 and tests with java11. This use case is more important. I don't think it is common to compile your pipeline with Java8 and then run with Java11.

Reply:
Thanks for input. I need to rewrite some logic then. Closing this PR for now